tool.go

```diff
@@ -498,7 +498,21 @@ func main() {
 	cmdServe.Run = func(cmd *cobra.Command, args []string) {
 		options.BuildTags = strings.Fields(*tags)
 		dirs := append(filepath.SplitList(build.Default.GOPATH), build.Default.GOROOT)
-		sourceFiles := http.FileServer(serveCommandFileSystem{options: options, dirs: dirs, sourceMaps: make(map[string][]byte)})
+		var root string
+		if len(args) == 1 {
```

Comment:
That sounds better, since it handles errors first. That is more in line with the recommended style, see https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow.